### DISCUSSION TITLE
What's missing?

### DISCUSSION THREAD

**Discussion by meza** [meza/trance-stack#63] [Ideas]
Please share ideas that you think should be part of this stack!

**Comment by docoder** [comment]
How to integrate databases ?

**Reply by meza** [reply]
We purposefully didn't add any databases to the stack because it's up to the app that you're building to add the one that you're using. There would be no point in me adding dynamo support when you'd want to use Postgres.

The [Remix docs](https://remix.run/docs/en/main/guides/data-loading#databases) however have a good explanation on how to integrate them.

**Reply by docoder** [reply]
For local development, do we need to use ARC for the database, and how do we sync with AWS?

**Comment by docoder** [comment]
How to integrate tailwindcss？

**Reply by meza** [reply]
It's a Remix stack, so the [Remix docs](https://remix.run/docs/en/main/guides/styling#tailwind-css) should help you with adding tailwind if you wish to

**Reply by jarimustonen** [reply]
Even if it is not supported, I think it should be mentioned in `postcss.config.js`. Specifically it would be nice to know which plugins are compatible with `tailwindcss` and if one chooses to include `tailwindcss` to `postcss.config.js` what would be the correct order.

**Comment by docoder** [comment]
Why not Chromatic?

**Reply by meza** [reply]
Good question! I've been thinking about adding it to the stack. I think once all the other issues are ironed out, that might be the next thing to add

**Comment by docoder** [comment]
Can it be used for multi-tenant applications with customizable domain names?

**Reply by meza** [reply]
Not without some modifications to the deployment but it's not too far off. You'd need to modify where the app gets the fundamental configuration from, like the Auth0 details - which should depend on the tenant.

**Comment by docoder** [comment]
Why not use Storybook's [test-runne](https://github.com/storybookjs/test-runner)r for testing?

**Reply by meza** [reply]
I haven't played with it yet but it seems very fun! I watched a demo on the [Storybook day](https://storybook.js.org/day/stage/main).

Will definitely play with it and see how it fits into the ecosystem. My worry is that the more tools we introduce to do _very similar_ things, it becomes less and less clear for new people to understand what goes where.

Playwright is also introducing [component tests](https://playwright.dev/docs/test-components) which is also very intriguing.

If anyone has any good written guidelines about what should live where and when, that would be a big help!

**Comment by docoder** [comment]
Why not sst.dev?

**Reply by meza** [reply]
Because sst is nothing more than a collection of CDK constructs that is managed by SST instead of you. If you want to modify it to suit your organisation's need, you'll end up back at what we have in this stack.

**Comment by miguelocarvajal** [comment]
Adding prettier would be cool as it could automatically clean up a lot of the stuff that the linter complains about.

**Reply by meza** [reply]
That's something you'll need to do locally for your own projects.

**Comment by jarimustonen** [comment]
What technologies to use for form handling and validation?

**Reply by meza** [reply]
Ah, great question! The stack itself doesn't have an opinion on this, it's up to you to implement your own way of dealing with forms.

However, I can recommend https://github.com/edmundhung/conform which is pretty cool. It has some issues with supporting the no-js path but it's still doable.

**Reply by jarimustonen** [reply]
Thank you. I will check that. I'm trying to wrestle with forms that progressively change based on fields. Validation frameworks seem not to like that.

Also, thank you for a great site! This is invaluable for those of us who are new to all things Remix. I last did web development eight years ago, and a lot has happened since. The only problem with your site/repo is that it is hard to find. I spotted it on a Stackoverflow discussion.
